projects/step-frontend/src/lib/modules/timeseries/time-series.service.ts

```diff
@@ -3,14 +3,21 @@ import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { FindBucketsRequest } from './find-buckets-request';
 import { TimeSeriesChartResponse } from './time-series-chart-response';
-import { AsyncTaskStatusResource, Execution } from '@exense/step-core';
+import { AsyncTaskStatusResource, AuthService, Execution } from '@exense/step-core';
+import { TimeSeriesConfig } from './time-series.config';
 
 // TODO provide it it TimeSeriesModule
 @Injectable({
   providedIn: 'root',
 })
 export class TimeSeriesService {
-  constructor(private http: HttpClient) {}
+  private readonly resolutionPeriod: number; // this is received from the backend
+
+  constructor(private http: HttpClient, private authService: AuthService) {
+    this.resolutionPeriod =
+      parseInt(authService.getConf()?.miscParams[TimeSeriesConfig.SERVICE_RESOLUTION_CONFIG_PROPERTY]!) ||
```

Comment:
parseInt looks incorrect. I'm not 100% sure what it does, but in TS you would usually use the Number constructor:
Number(authService.getConf()?.miscParams[TimeSeriesConfig.SERVICE_RESOLUTION_CONFIG_PROPERTY]!)

Also it looks weird that you use ? to say getConf() might be undefined but then guarantee that there is a value with the closing !. In general using the exclamation mark without checks should be avoided except if you are 100% sure it will always work, then you can replace the ? also with one.

Reply:
@neogucky 
`parseInt` takes string and tries to parse it and return number.
But if the string was incorrect, it returns NaN.
I think it's okay to use `parseInt` here, but it's result should be additional checked with `isNaN` function

Reply:
> @neogucky `parseInt` takes string and tries to parse it and return number. But if the string was incorrect, it returns NaN. I think it's okay to use `parseInt` here, but it's result should be additional checked with `isNaN` function

I still prefer Number since parseInt is extremely error tolerant, something we shouldn't need in our own eco-system: https://thisthat.dev/number-constructor-vs-parse-int/#:~:text=Differences,parses%20the%20value%20of%20input.&text=As%20you%20see%2C%20parseInt%20will,to%20convert%20the%20entire%20string.